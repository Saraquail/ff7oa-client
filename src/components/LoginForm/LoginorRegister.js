import React, { Component } from 'react'
import { HashLink } from 'react-router-hash-link';
import { Link } from 'react-router-dom'

import LoginForm from './LoginForm'
import RegisterForm from './RegisterForm'

class LoginorRegister extends Component {
  state = {
    login: false,
    register: false
  }

  handleLogin = ev => {
    this.setState({
      login: true,
      register: false
    })
  }

  handleRegister = ev => {
    this.setState({
      register: true,
      login: false
    })
  }

  render () {
    return(
      <section className="LoginOrRegister">
        <div>
        <Link to="/bestiary">
          <button>Continue as a guest</button>
        </Link>
        </div>
        <HashLink to="#register-header">
        <button className="login-button"
        onClick={this.handleRegister}>
        Register
        </button>
        </HashLink>
        <HashLink to="#login-header">
          <button className="login-button"
            onClick={this.handleLogin}>
            Login
          </button>
        </HashLink>
        {this.state.login ? <LoginForm /> : ''}
        {this.state.register ? <RegisterForm /> : ''}
      </section>
    
    )
  }
}

export default LoginorRegister