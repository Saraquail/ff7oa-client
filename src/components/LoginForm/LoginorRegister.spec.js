import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import LoginorRegister from './LoginorRegister.js';

describe('<LoginorRegister/>', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <BrowserRouter>
        <LoginorRegister />
      </BrowserRouter>, div,
    );

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <LoginorRegister />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
