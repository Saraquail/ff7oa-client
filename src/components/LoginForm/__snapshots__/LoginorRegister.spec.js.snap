// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginorRegister/> renders the UI as expected 1`] = `
<section
  className="LoginOrRegister"
>
  <div>
    <a
      href="/bestiary"
      onClick={[Function]}
    >
      <button>
        Continue as a guest
      </button>
    </a>
  </div>
  <a
    href="/#register-header"
    onClick={[Function]}
  >
    <button
      className="login-button"
      onClick={[Function]}
    >
      Register
    </button>
  </a>
  <a
    href="/#login-header"
    onClick={[Function]}
  >
    <button
      className="login-button"
      onClick={[Function]}
    >
      Login
    </button>
  </a>
  
  
</section>
`;
